dcpy/connectors/edm/publishing.py

```diff
@@ -230,13 +246,59 @@ def legacy_upload(
             s3.copy_file(BUCKET, str(key), str(prefix / "latest" / output.name), acl)
 
 
+def upload_build(
+    output_path: Path,
+    product: str,
+    *,
+    acl: s3.ACL,
+    build: str | None = None,
+    max_files: int = s3.MAX_FILE_COUNT,
+) -> BuildKey:
+    """
+    Uploads a product build to an S3 bucket.
+
+    This function handles uploading a local output folder to a specified
+    location in an S3 bucket. The path, product, and build name must be
+    provided, along with an optional ACL (Access Control List) to control
+    file access in S3.
+
+    Raises:
+        FileNotFoundError: If the provided output_path does not exist.
+        ValueError: If the build name is not provided and cannot be found in the environment variables.
+    """
+    if not output_path.exists():
+        raise FileNotFoundError(f"Path {output_path} does not exist")
+    build_name = build or BUILD_NAME
+    if not build_name:
+        raise ValueError(
+            f"Build name supplied via CLI or the env var 'BUILD_NAME' cannot be '{build_name}'."
+        )
+    build_key = BuildKey(product, build_name)
+    logger.info(f'Uploading {output_path} to {build_key.path} with ACL "{acl}"')
+    upload(output_path, build_key, acl=acl, max_files=max_files)
+
+    version = get_version(build_key)
+    event_metadata = EventLog(
+        event_type=EventType.BUILD,
+        product=build_key.product,
+        release_version=version,
+        new_path=build_key.path,
+        old_path=None,
+        run_details=metadata.get_run_details(),
+    )
+    if build_key.build not in IGNORED_LOGGING_BUILDS:
+        log_event_in_db(event_metadata)
```

Comment:
(nits) I think this could all go under the `if build_key.build not in IGNORED_LOGGING_BUILDS:` clause

I was gonna suggest `EventLog` take a `ProductKey` as an argument to reduce the number of args and having to do `version = get_version(build_key)`. but maybe we don't want models to be dependent on other models?

Reply:
although, I do see now that `dcpy/models/lifecycle/builds.py` imports other models

Reply:
Yeah, I was thinking of this approach initially. But I ultimately decided to set up `EventLog` like this because: 1) `promote_to_draft` and `publish` steps need both "old" and "new" Product Keys and 2) having external interactions with s3 or env variables (i.e. getting a version from s3 for `build` step or getting `run_details` from envars) would add more complexity.